daffodil-lib/src/main/scala/org/apache/daffodil/xml/DaffodilSAXParserFactory.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.xml
+
+import org.apache.xerces.jaxp.SAXParserFactoryImpl
+
+
+class DaffodilSAXParserFactory private ()
+  extends SAXParserFactoryImpl() {
+
+  override def newSAXParser() = {
+    val res = super.newSAXParser()
+    XMLUtils.setSecureDefaults(res.getXMLReader)
```

Comment:
Does this make it so all future calls to getXMLReader will have secure defaults set? I assumed that getXMLReader would allocate a new instance everytime it's call, and so would need setSecureDefaults set on it?

Reply:
Looking at the Xerces code, I'm wrong. A SAXParser does have only a single xmlReader, and getXMLReader always returns that same reader. So what you have is correct.

Reply:
Yeah, it's a getter, not an allocator. But I can also override so that getXMLReader is our override, and then it's 100% clear.